### PR TITLE
Advanced Baton Refactor and Probe Mode Buff

### PR DESCRIPTION
**Pull request by Super3222**
### Intent of your Pull Request

Refactored the advanced baton code specifically for a weaker variants, may come in handy for https://github.com/yogstation13/yogstation/pull/917.

Probe mode can now detect cultists, shadowlings, thralls, and cybermen. 
#### Changelog

:cl:
rscadd: An unknown race of alien abductors have been studying SS13 for quite a while now. They've upgraded their technology, specifically the advanced baton, so it can detect cultists strung to elder gods, shadowlings, their thralls, and cybermen.
/:cl:

### CODE REVIEW

**Comment by ShadowDeath6** [comment]
lmao we're not having an antag checker for science if adam's pr does pass

File: code/game/gamemodes/miniantags/abduction/abduction_gear.dm [hunk 1]
```diff
@@ -492,6 +495,22 @@ Congratulations! You are now trained for xenobiology research!"}
 		species = "<span clas=='notice'>[H.dna.species.name]</span>"
 		if(L.mind && L.mind.changeling)
 			species = "<span class='warning'>Changeling lifeform</span>"
+
+		if(L.mind && L.mind.special_role == "Servant of Ratvar")
+			species = "<span class='warning'>Lifeform has connections to the elder god, Ratvar.</span>"
+
+		if(L.mind && L.mind.special_role == "Cultist")
+			species = "<span class='warning'>Lifeform has connections to the elder god, Nairse.</span>"
```
Comment (ShadowDeath6) [inline code/game/gamemodes/miniantags/abduction/abduction_gear.dm]: Nar'Sie, fuck off with nairse

**Comment by Super3222** [comment]
That's the point of having an artificial version, Shadow.
EDIT: One that CAN'T probe like the real version.

File: code/game/gamemodes/miniantags/abduction/abduction_gear.dm (hunk 1 above)
Comment (Super3222) [reply code/game/gamemodes/miniantags/abduction/abduction_gear.dm]: nairse

Comment (ghost) [reply code/game/gamemodes/miniantags/abduction/abduction_gear.dm]: Rat'Var

Nar'Sie

FUCK WIT

Comment (Super3222) [reply code/game/gamemodes/miniantags/abduction/abduction_gear.dm]: the abductors couldn't get past the apostrophes so they skipped them

**Comment by Kmc2000** [comment]
Make the probe only worked when aimed at the groin and we good

**Comment by Super3222** [comment]
Merge ready

**Pull request merged**